### DISCUSSION TITLE
How to generate a single docs/openapi.json for multiple FastAPi-based Docker containers?

### DISCUSSION THREAD

**Discussion by SmartDataCode** [fastapi/fastapi#4163] [Questions]
Hi,

I'm building an application that consists of a number of microservices, each developed using FastApi and each one deployed in a seperate Docker container. Depending on the workload, I can scale up or down the number of containers on one or more microservices. I'm using Traefik as router/load balancer.
Currently, I have the API docs seperate for each container, like (as an example):
- www.example.com/service-1/docs
- www.example.com/service-2/docs
- www.example.com/service-3/docs
- etc.

What I would like to have though, is a single API docs page, where all endpoints are listed together. In other words: how do I integrate the docs of all the individual FastApi apps into a single docs page, like
- www.example.com/docs (with sections service-1, service-2, etc)

Anybody a clue?

**Comment by froi** [comment]
I don't know if this helps you and I haven't tested this, but [Redoc's OpenAPI-CLI has a join function](https://redoc.ly/docs/cli/commands/join/) that might help you. Once you have the OpenAPI spec you can generate you bundle HTML or Swagger docs and host them somewhere.

Maybe have this be part of your CI?
